src/modules/Game.class.js

```diff
@@ -1,68 +1,269 @@
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
-  }
-
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
-
-  /**
-   * @returns {number}
-   */
-  getScore() {}
-
-  /**
-   * @returns {number[][]}
-   */
-  getState() {}
-
-  /**
-   * Returns the current game status.
-   *
-   * @returns {string} One of: 'idle', 'playing', 'win', 'lose'
-   *
-   * `idle` - the game has not started yet (the initial state);
-   * `playing` - the game is in progress;
-   * `win` - the game is won;
-   * `lose` - the game is lost
-   */
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
+  static gameStatuses = {
+    idle: 'idle',
+    playing: 'playing',
+    win: 'win',
+    lose: 'lose',
+  };
+
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.initialState = initialState;
+    this.state = this.initialState.map((row) => [...row]);
+    this.status = Game.gameStatuses.idle;
+    this.score = 0;
+  }
+
+  moveLeft() {
+    if (this.getStatus() !== Game.gameStatuses.playing) {
+      return;
+    }
+
+    const updatedState = this.state.map((row) => this.move(row));
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(updatedState)) {
+      this.updateGameState(updatedState);
+      this.addCells();
+    }
+  }
+
+  moveRight() {
+    if (this.getStatus() !== Game.gameStatuses.playing) {
+      return;
+    }
+
+    const reverseState = this.state.map((row) => [...row].reverse());
+
+    const updatedState = reverseState.map((row) => {
+      return this.move(row).reverse();
+    });
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(updatedState)) {
+      this.updateGameState(updatedState);
+      this.addCells();
+    }
+  }
+
+  moveUp() {
+    if (this.getStatus() !== Game.gameStatuses.playing) {
+      return;
+    }
+
+    const rotateState = this.rotateMatrixCounteClockwise(this.getState());
+
+    const updatedState = rotateState.map((row) => this.move(row));
+
+    const unRotateSteate = this.rotateMatrixClockwise(updatedState);
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(unRotateSteate)) {
+      this.updateGameState(unRotateSteate);
+      this.addCells();
+    }
+  }
+
+  moveDown() {
+    if (this.getStatus() !== Game.gameStatuses.playing) {
+      return;
+    }
+
+    const rotateState = this.rotateMatrixClockwise(this.getState());
+
+    const updatedState = rotateState.map((row) => this.move(row));
+
+    const unRotateSteate = this.rotateMatrixCounteClockwise(updatedState);
+
+    if (JSON.stringify(this.getState()) !== JSON.stringify(unRotateSteate)) {
```

Comment:
Consider refactoring the state comparison into a single reusable function to avoid code duplication.